controllers/activemqartemis_reconciler.go

```diff
@@ -820,7 +921,22 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) ExposureDefinitionForCR(customR
 }
 
 func (reconciler *ActiveMQArtemisReconcilerImpl) trackDesired(desired rtclient.Object) {
-	reconciler.requestedResources = append(reconciler.requestedResources, desired)
+	desiredType := reflect.TypeOf(desired)
+	if reconciler.requestedResources == nil {
+		reconciler.requestedResources = make(map[reflect.Type]map[string]rtclient.Object)
+	}
+	resMap, ok := reconciler.requestedResources[desiredType]
+	if !ok {
+		resMap = make(map[string]rtclient.Object)
+		reconciler.requestedResources[desiredType] = resMap
+	}
+	resName := desired.GetName()
+	_, ok = resMap[resName]
+	if !ok {
+		resMap[resName] = desired
+	} else {
+		reconciler.log.V(1).Info("resource already tracked", "resource", desired)
```

Comment:
am wondering about this logging, does it imply that we are doing a multi phase reconcile, as in not just one pass. Why would code track a resource twice? it may be indicative of some logic error possibly?
Also, should last in not win? if we do track twice for some good reason.

Reply:
Yeah I think track a same resource twice may imply some logic error. Even there is some such case with good reason the last should win.